metadata_manager.py

```diff
@@ -58,8 +58,59 @@ def get_all_server(self, local):
 
     # Clear all metadata from the database.
     def clear_metadata(self):
-        self.cursor.execute('DELETE FROM Server WHERE 1=1')
+        self.cursor.execute('DELETE FROM KnownServer WHERE 1=1')
         self.cursor.execute('DELETE FROM FileMap WHERE 1=1')
+        self.cursor.execute('DELETE FROM Stats WHERE 1=1')
+        self.conn.commit()
+
+    # Adds a concurrent request of a uuid
+    def add_concurrent_request(self, uuid):
```

Comment:
I think.. both this and `remove_concurrent_request` are non-atomic. If I have a situation where two processes call this at the same time, it is possible to have a condition where the `UPDATE` will overwrite another.

* Process A READ 0
* Process B READ 0
* Process A UPDATE 1
* Process B UPDATE 1

What might be a solution to this is to have the `Stats` or `Connections` table (whatever you want to call it), store a schema with `uuid text, client_ip text, timestamp int`, where `timestamp` is the time of initiating the connection and have an individual record for each client's connection to the server. (This takes into account that a single client can have two connections to the server for the same file, though I don't know why that would ever really happen.)

Then, in `get_concurrent_request`, you can run a `COUNT` query to retrieve the number of concurrent requests.